dotCMS/src/main/java/com/dotcms/rest/api/v1/asset/WebAssetResource.java

```diff
@@ -49,7 +66,186 @@ public Response getAssetsInfo(@Context final HttpServletRequest request,
+
+        final InitDataObject initDataObject = new WebResource.InitBuilder()
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true).init();
+
+        final User user = initDataObject.getUser();
+
+        final WebAssetView webAssetView = helper.saveUpdateAsset(request, form, user);
+        Logger.info(this, String.format("User [%s] is uploading asset for path [%s]", user.getUserId(), form.getAssetPath()));
+        return Response.ok(new WebAssetEntityView(webAssetView)).build();
+    }
+
+    /**
+     * Delete an asset by path
+     * @param request
+     * @param response
+     * @param form
+     * @return
+     * @throws DotSecurityException
+     * @throws DotDataException
+     * @throws IOException
+     */
+    @Path("/")
+    @DELETE
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response deleteAsset(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            AssetInfoRequestForm form
+    ) throws DotSecurityException, DotDataException, IOException {
+
+        final InitDataObject initDataObject = new WebResource.InitBuilder()
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true).init();
+
+        final User user = initDataObject.getUser();
+        helper.deleteAsset(form.assetPath(), user);
+        Logger.info(this,
+                String.format("User [%s] deleted asset for path [%s] ", user.getUserId(), form.assetPath()));
+        return Response.ok().build();
+    }
+
+
+    /**
+     * Delete an asset by path
+     * @param request
+     * @param response
+     * @param form
+     * @return
+     * @throws DotSecurityException
+     * @throws DotDataException
+     * @throws IOException
+     */
+    @Path("/archive")
+    @DELETE
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response archiveAsset(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            AssetInfoRequestForm form
+    ) throws DotSecurityException, DotDataException, IOException {
+
+        final InitDataObject initDataObject = new WebResource.InitBuilder()
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true).init();
+
+        final User user = initDataObject.getUser();
+        helper.archiveAsset(form.assetPath(), user);
+        Logger.info(this,
+                String.format("User [%s] deleted asset for path [%s] ", user.getUserId(), form.assetPath()));
+        return Response.ok().build();
```

Comment:
you may return here ResponseEntityBooleanView in order to have typed responses for swagger